Main-rpn/Main-rpn/Main-rpn.cpp

```diff
@@ -0,0 +1,97 @@
+#include <iostream>
+#include <fstream>
+
+#include <math.h>
+#include <string>
+#include <stack>
+
+int calculate(const std::string &infix)
+{
+    std::cout << infix << "\n";
+    std::string postfix = "";
+
+    std::stack<char> operators;
+
+    // Convert from infix to postfix notation
+    for (int i = 0; i < infix.size(); i++)
+    {
+        if (infix[i] >= 48 && infix[i] <= 57)
+        {
+            postfix += infix[i];
+            postfix += ' ';
+        }
+        else if (infix[i] == '+' || infix[i] == '-' || infix[i] == '*' || infix[i] == '/' || infix[i] == '^')
+            operators.push(infix[i]);
+        if (infix[i] == ')' || i == infix.size() - 1)
+            if (!operators.empty())
+            {
+                postfix += operators.top();
+                postfix += ' ';
+                operators.pop();
+            }  
+    }
+
+    // Calculate the result from the postfix notation
+    // Assume all numbers are 1-digit 
+
+    std::stack<int> operands;
+
+    for (int i = 0; i < postfix.size(); i++)
+    {
+        if (postfix[i] >= 48 && postfix[i] <= 57)
+            operands.push(int(postfix[i]) - 48);
+        if (postfix[i] == '+' || postfix[i] == '-' || postfix[i] == '*' || postfix[i] == '/' || postfix[i] == '^')
+        {
+            int firstNum;
+            int secondNum;
+            if (!operands.empty())
+            {
+                secondNum = operands.top();
+                operands.pop();
+            }
+            if (!operands.empty())
+            {
+                firstNum = operands.top();
+                operands.pop();
+            }
+
+            if (postfix[i] == '+')
+                operands.push(firstNum + secondNum);
+            if (postfix[i] == '-')
+                operands.push(firstNum - secondNum);
+            if (postfix[i] == '*')
+                operands.push(firstNum * secondNum);
+            if (postfix[i] == '/')
+                operands.push(firstNum / secondNum);
+            if (postfix[i] == '^')
+                operands.push(pow(firstNum, secondNum));
```

Comment:
Please explicitly cast here to notify that the case from double to int is intended. `operands.push((int)pow(firstNum, secondNum));` https://rules.sonarsource.com/cpp/RSPEC-5276